modules/repository-s3/src/javaRestTest/java/org/elasticsearch/repositories/s3/AbstractRepositoryS3RestTestCase.java

```diff
@@ -0,0 +1,389 @@
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.cluster.routing.Murmur3HashFunction;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.repositories.blobstore.BlobStoreRepository;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.junit.Before;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+
+public abstract class AbstractRepositoryS3RestTestCase extends ESRestTestCase {
+
+    public record TestRepository(String repositoryName, String clientName, String bucketName, String basePath) {
+
+        public Closeable register() throws IOException {
+            return register(UnaryOperator.identity());
+        }
+
+        public Closeable register(UnaryOperator<Settings> settingsUnaryOperator) throws IOException {
+            assertOK(client().performRequest(getRegisterRequest(settingsUnaryOperator)));
+            return () -> assertOK(client().performRequest(new Request("DELETE", "/_snapshot/" + repositoryName())));
+        }
+
+        private Request getRegisterRequest(UnaryOperator<Settings> settingsUnaryOperator) throws IOException {
+            return newXContentRequest(
+                HttpMethod.PUT,
+                "/_snapshot/" + repositoryName(),
+                (b, p) -> b.field("type", S3Repository.TYPE)
+                    .startObject("settings")
+                    .value(
+                        settingsUnaryOperator.apply(
+                            Settings.builder()
+                                .put("bucket", bucketName())
+                                .put("base_path", basePath())
+                                .put("client", clientName())
+                                .put("canned_acl", "private")
+                                .put("storage_class", "standard")
+                                .put("disable_chunked_encoding", randomBoolean())
+                                .build()
+                        )
+                    )
+                    .endObject()
+            );
+        }
+    }
+
+    protected abstract String getBucketName();
+
+    protected abstract String getBasePath();
+
+    protected abstract String getClientName();
+
+    protected static String getIdentifierPrefix(String testSuiteName) {
+        return testSuiteName + "-" + Integer.toString(Murmur3HashFunction.hash(testSuiteName + System.getProperty("tests.seed")), 16) + "-";
+    }
+
+    @Before
+    public void skipIfFips() {
+        assumeFalse("doesn't work in a FIPS JVM, but that's ok", inFipsJvm());
```

Comment:
Is this still necessary? The existing yaml tests do not seem to need it. But we can also tackle this in a separate PR.

Reply:
I think you're right now that you've fixed the fixtures to make long enough keys in #117675. Removed.